src/backend/distributed/planner/recursive_planning.c

```diff
@@ -599,6 +611,267 @@ RecursivelyPlanNonColocatedSubqueriesInWhere(Query *query,
 }
 
 
+/*
+ * RecursivelyPlanRecurringLeftJoins
+ */
+static void
+RecursivelyPlanRecurringLeftJoins(Node *node,
```

Comment:
Overall, we seem to have two main classes of limitations in this draft: 


```SQL
-- inner part is another JOIN tree
select count(*) FROM 
users_table_ref 
   LEFT JOIN 
(users_table JOIN events_table USING (user_id)) USING (user_id);

-- inner part is a subquery/cte
select count(*) FROM 
users_table_ref 
   LEFT JOIN 
(SELECT * FROM users_table) u1 USING (user_id);
```

For the latter, can we do:
```C
index 6286290b7..70e227d4a 100644
--- a/src/backend/distributed/planner/recursive_planning.c
+++ b/src/backend/distributed/planner/recursive_planning.c
@@ -879,6 +879,11 @@ RecursivelyPlanNonRecurringJoinNode(Node *distributedNode, Query *query,
                ReplaceRTERelationWithRteSubquery(distributedRte, requiredAttributes,
                                                                                  recursivePlanningContext);
        }
+       else if (distributedRte->rtekind == RTE_SUBQUERY)
+       {
+
+               RecursivelyPlanSubquery(distributedRte->subquery, recursivePlanningContext);
+       }
        else

```

For the former, I'm still thinking..